### PR TITLE
[WIP] Granular import APIs

### PR DESCRIPTION
**Pull request by aronasorman**
# Checklist

- [x] PR has the correct target milestone when it's merged
- [x] PR has been fully tested manually
- [ ] Documentation is updated as necessary
- [ ] External dependency files are updated (`yarn` and `pip`)
- [ ] If internal dependency are updated, link to diff is included
- [ ] Screenshots of any front-end changes are in the PR description
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] You've added yourself to AUTHORS.rst if you're not there

# TODO
- Create a new `importchannel` endpoint that simply imports a channel DB to the local machine.
- Limit the `localimport` and `networkimport` endpoints to just downloading the content
- Add the host parameter to the `importchannel` and `networkimport` endpoints

# Details

This PR adds the following features to content importing:

- Include an optional `host` parameter to the `importcontent` and `importchannel` management commands. This allows a user to download content from another server apart from the official studio server.
- Include an optional `--node_ids` parameter to `importcontent`. When called through the command line, you'll need to pass in a string composed of each node id you want to download, separated by a comma, e.g. (`--node_ids=<id1>,<id2>,<id3>`). When called programmatically, you just need to pass in a list of node ids. `importcontent` will then only download nodes that have those ids passed in.
- Extend the `networkimport` and `localimport` REST APIs to accept a `node_ids` key, with the value being a list of node ids the user wishes to download.
- Upgrades Iceqube to 0.0.4, adding support for arbitrary metadat. [Diff is here](https://github.com/learningequality/iceqube/pull/28).

### CODE REVIEW

File: kolibri/content/management/commands/importcontent.py [hunk 1]
```diff
@@ -23,6 +23,26 @@ def add_arguments(self, parser):
...
+        Make sure to call this last in the argument list.
+
+        e.g.
+
+        kolibri manage importcontent network <channel id> --node_ids <id1> <id2> [<ids>...]
+        """
+        parser.add_argument(
+            "--node_ids", "-n",
+            # Split the comma separated string we get, into a list of strings
+            type=lambda x: x.split(","),
```
Comment (rtibbles) [inline kolibri/content/management/commands/importcontent.py:39]: Could you just use the `nargs="+"` keyword here to allow multiple values? https://docs.djangoproject.com/en/1.11/howto/custom-management-commands/#module-django.core.management

File: kolibri/tasks/api.py [hunk 2]
```diff
@@ -120,14 +121,22 @@ def startlocalimport(self, request):
...
+                "The 'drive_id' field is required."
+            )
+
+        if "channel_id" not in request.data:
+            raise serializers;ValidationError(
+                "The 'channel_id' field is required."
+            )
+
+        channel_id = request.data["channel_id"]
+        node_ids = request.data.get("node_ids")
```
Comment (rtibbles) [inline kolibri/tasks/api.py]: If you use my suggestion above, you would probably have to do the split here though!

**Comment by aronasorman** [comment]
There's still work to be done here. I'll be everything listed in the [CLI portion of this doc](https://docs.google.com/document/d/1FGR4XBEu7IbfoaEy-8xbhQx2PvIyxp0VugoPrMfo4R4/edit#heading=h.3umyx76m52u7), into this PR.

**Comment by aronasorman** [comment]
@ralphiee22 I've added `startremotechannelimport`, `startremotecontentimport`, and `startdiskcontentimport`, which hook into the CLI stuff you're doing. Once you've pushed that, please test it so we can see if the API now works.

**Comment by lyw07** [comment]
Getting an error when running `kolibri manage importcontent network 8b4d3e6d3d4842ba8ea658335b5dd252` after successfully running `kolibri manage importchannel network 8b4d3e6d3d4842ba8ea658335b5dd252`. 

> Traceback (most recent call last):
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 365, in execute
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 293, in _wrap_coverage
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 325, in _wrap_profiling
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 408, in _execute
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 466, in execute_entry
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 471, in execute_module
  File "/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 192, in run_module
    fname, loader, pkg_name)
  File "/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/__main__.py", line 8, in <module>
    main(args=sys.argv[1:])
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/utils/cli.py", line 616, in main
    manage(command, args=django_args)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/utils/cli.py", line 447, in manage
    execute_from_command_line(argv=argv)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/tasks/management/commands/base.py", line 99, in handle
    return self.handle_async(*args, **options)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/management/commands/importcontent.py", line 185, in handle_async
    self.download_content(options["channel_id"], node_ids=options["node_ids"], base_url=options["base_url"])
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/management/commands/importcontent.py", line 106, in download_content
    self._transfer(DOWNLOAD_METHOD, channel_id, node_ids=node_ids, exclude_node_ids=exclude_node_ids, base_url=base_url)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/management/commands/importcontent.py", line 124, in _transfer
    total_bytes_to_transfer = files_to_download.aggregate(Sum('file_size'))['file_size__sum'] or 0
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/query.py", line 358, in aggregate
    return query.get_aggregation(self.db, kwargs.keys())
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/query.py", line 447, in get_aggregation
    outer_query.add_subquery(inner_query, using)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/subqueries.py", line 217, in add_subquery
    subquery=True,
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/compiler.py", line 389, in as_sql
    where, w_params = self.compile(self.where) if self.where is not None else ("", [])
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/compiler.py", line 366, in compile
    sql, params = node.as_sql(self, self.connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/where.py", line 79, in as_sql
    sql, params = compiler.compile(child)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/compiler.py", line 366, in compile
    sql, params = node.as_sql(self, self.connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/fields/related_lookups.py", line 84, in as_sql
    return super(RelatedIn, self).as_sql(compiler, connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 229, in as_sql
    return super(In, self).as_sql(compiler, connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 158, in as_sql
    rhs_sql, rhs_params = self.process_rhs(compiler, connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 216, in process_rhs
    sqls, sqls_params = self.batch_process_rhs(compiler, connection, rhs)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 52, in batch_process_rhs
    self.lookup_name, rhs, connection, prepared=True)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/fields/__init__.py", line 779, in get_db_prep_lookup
    prepared=prepared) for v in value]
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/fields/related.py", line 918, in get_db_prep_value
    return self.target_field.get_db_prep_value(value, connection, prepared)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/models.py", line 62, in get_db_prep_value
    value = uuid.UUID(value)
  File "/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/uuid.py", line 136, in __init__
    raise ValueError('badly formed hexadecimal UUID string')
ValueError: badly formed hexadecimal UUID string

**Comment by lyw07** [comment]
Getting an error when running `kolibri manage importcontent network 8e39cb56309745199ffb6b3e7a482d26` after successfully running `kolibri manage importchannel network 8e39cb56309745199ffb6b3e7a482d26`. 

But when `--node_ids` is not omitted, it works fine. This behaviors seems to be not consistent with the first comment in the [spec doc](https://docs.google.com/document/d/1FGR4XBEu7IbfoaEy-8xbhQx2PvIyxp0VugoPrMfo4R4/edit?ts=59fbaba5#heading=h.d9o09td26sdq)

> Traceback (most recent call last):
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 365, in execute
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 293, in _wrap_coverage
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 325, in _wrap_profiling
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 408, in _execute
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 466, in execute_entry
  File "/Users/lingyiwang/Downloads/kolibri%2Fbuildkite%2Fbuild-2517%2F3361%2Fkolibri-0.7.dev020171107170204-git.pex/.bootstrap/_pex/pex.py", line 471, in execute_module
  File "/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 192, in run_module
    fname, loader, pkg_name)
  File "/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/__main__.py", line 8, in <module>
    main(args=sys.argv[1:])
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/utils/cli.py", line 616, in main
    manage(command, args=django_args)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/utils/cli.py", line 447, in manage
    execute_from_command_line(argv=argv)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/__init__.py", line 353, in execute_from_command_line
    utility.execute()
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/__init__.py", line 345, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/base.py", line 348, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/tasks/management/commands/base.py", line 99, in handle
    return self.handle_async(*args, **options)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/management/commands/importcontent.py", line 185, in handle_async
    self.download_content(options["channel_id"], node_ids=options["node_ids"], base_url=options["base_url"])
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/management/commands/importcontent.py", line 106, in download_content
    self._transfer(DOWNLOAD_METHOD, channel_id, node_ids=node_ids, exclude_node_ids=exclude_node_ids, base_url=base_url)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/management/commands/importcontent.py", line 124, in _transfer
    total_bytes_to_transfer = files_to_download.aggregate(Sum('file_size'))['file_size__sum'] or 0
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/query.py", line 358, in aggregate
    return query.get_aggregation(self.db, kwargs.keys())
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/query.py", line 447, in get_aggregation
    outer_query.add_subquery(inner_query, using)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/subqueries.py", line 217, in add_subquery
    subquery=True,
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/compiler.py", line 389, in as_sql
    where, w_params = self.compile(self.where) if self.where is not None else ("", [])
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/compiler.py", line 366, in compile
    sql, params = node.as_sql(self, self.connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/where.py", line 79, in as_sql
    sql, params = compiler.compile(child)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/sql/compiler.py", line 366, in compile
    sql, params = node.as_sql(self, self.connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/fields/related_lookups.py", line 84, in as_sql
    return super(RelatedIn, self).as_sql(compiler, connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 229, in as_sql
    return super(In, self).as_sql(compiler, connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 158, in as_sql
    rhs_sql, rhs_params = self.process_rhs(compiler, connection)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 216, in process_rhs
    sqls, sqls_params = self.batch_process_rhs(compiler, connection, rhs)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/lookups.py", line 52, in batch_process_rhs
    self.lookup_name, rhs, connection, prepared=True)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/fields/__init__.py", line 779, in get_db_prep_lookup
    prepared=prepared) for v in value]
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/dist/django/db/models/fields/related.py", line 918, in get_db_prep_value
    return self.target_field.get_db_prep_value(value, connection, prepared)
  File "/Users/lingyiwang/.pex/install/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl.5e2bbf6b21207323a09f0612d3e64ddf01a33925/kolibri-0.7.dev020171107170204_git-py2.py3-none-any.whl/kolibri/content/models.py", line 62, in get_db_prep_value
    value = uuid.UUID(value)
  File "/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/uuid.py", line 136, in __init__
    raise ValueError('badly formed hexadecimal UUID string')
ValueError: badly formed hexadecimal UUID string

File: kolibri/tasks/api.py [hunk 3]
```diff
@@ -191,7 +280,7 @@ def localdrive(self, request):
         return Response(out)
 
 
-def _networkimport(channel_id, update_progress=None, check_for_cancel=None):
+def _networkimport(channel_id, node_ids, update_progress=None, check_for_cancel=None):
```
Comment (lyw07) [inline kolibri/tasks/api.py:308]: Do we still need `_networkimport`, `_localimport`, `_localexport`, `_deletechannel` functions?

Comment (aronasorman) [reply kolibri/tasks/api.py:308]: Not anymore -- i kept it in so we can reference it until it's fully implemented.

**Pull request merged**